### ISSUE TITLE
Only sources and console panels are themed

### ISSUE THREAD

**Issue by micjamking** [micjamking/devtools-author#31]
I've made a conscious design decision not to apply the selected theme in the current version of *DevTools Author* to all panels, but rather only the `Sources` and `Console` panels within DevTools. The reasons for this is [maintainability](#maintainability), [usability](#usability), and [consistency](#consistency).

### Maintainability
First off, skinning Chrome's Developer Tools is not a simple task. It requires debugging DevTools with DevTools (sometimes 3-4 levels deep), fighting with CodeMirror's own CSS specificity/inheritance, and using deprecated selectors (`::shadow` and `/deep/`) to pierce the shadow DOM and style components from the outside. 

All of this means that each time something changes or updates in DevTools (which is not infrequent), the project needs to be updated just to maintain the current experience. The smaller the footprint *DevTools Author* has, the less time spent maintaining the extension and the more time I have to focus on providing the best experience for the panels it does support.

### Usability
This may sound subjective, but I simply don't feel the usability of the other panels is improved by theming, and the exact opposite result (poor legibility/usability) is much more likely to happen by enabling the themes in panels outside of the authoring workflow. This brings me to my next point, which is to maintain consistency with the purpose of the extension and DevTools as a whole.

### Consistency
I created *DevTools Author* to solve a specific problem; increase legibility and improve the overall _authoring_ experience within DevTools, keyword being 'authoring'. Outside of the `Sources` and `Console` panels, very little 'authoring' actually takes place. The remaining panels, while interactive, are mostly read-only and used for auditing purposes, the single exception being the `Elements` panel once workspaces have been configured. 

`Elements` is the only panel I have or will consider adding support for due to its integration with the `Sources` panel. However, the current styling within the `Elements` panel has a higher purpose to stay in sync with the original source while reflecting the dynamic state of the markup, which can be seen when opening *View > Developer > View Source* in Chrome. This purpose is even reiterated in DevTools default CSS comments:

**ui/inspectorSyntaxHighlight.css**
```css
.webkit-html-doctype {
    /* Keep this in sync with view-source.css (.webkit-html-doctype) */
    color: rgb(192, 192, 192);
}
```
For consistency sake, I've simply chosen not to style the `Elements` panel via the theme at this time, however `font-size` and `font-family` settings from *DevTools Author* will apply here.

I may change my mind about this in the future if maintenance becomes any easier (which I doubt), but for the time being, I have no intention of supporting theming of panels outside of `Sources` and `Console`.

**Comment by okor** [comment]
So, just one human here - but I spend most of my day in the Network and Timeline tabs (I'm a perf engineer). I read how you created this with the original intent of focusing on the authoring experience. I get it. But for me, I would like to see the other tabs inherit the same styles.

Not unlike this project: https://chrome.google.com/webstore/detail/devtools-theme-zero-dark/bomhdjeadceaggdgfoefmpeafkjhegbo?hl=en-US but with all the wonderful themes you have provided. 

Just a vote. Totally understand if it's not a priority. Thanks!

**Comment by micjamking** [comment]
Hey @okor, thanks for reaching out! 

I totally get it, I promise you. Who knows, if this issue get's enough :+1: AND I (or someone) has the time to contribute, I'd be open to reconsider. 

DevTools Author started out as a pet project that worked fine for JUST ME, but now that its been released, I'm no longer the primary audience, and so I'm genuinely open to feedback and criticism.

That being said, Chrome v50 (Canary) is shipping with a light/dark theme switcher which effectively skins the entire DevTools window from end-to-end, every panel. My future intentions for DevTools Author is to continue overriding the main authoring panels (elements, console, sources) while allowing the built-in light/dark theme to cover the remaining non-editable panels. The intent and focus of the plugin would remain the same, while also taking advantage of the built-in styling to provide a more comprehensive "theme".

I'm currently in the process of fixing bugs/issues in Canary in preparation, but it's still not quite there yet for the sources panel. Once this is complete, you can checkout the Canary release channel and see if it accommodates your needs.

**Comment by okor** [comment]
@micjamking Oh, I hadn't seen the dark theme switcher in Canary yet, that's rad. In that case, since a lot of color/contrast edge cases are probably handled by the default "dark" theme, I would agree that it's probably not worth the effort to do a bunch of work on the Network and other panels at this time. I definitely like have this plugin so I can modify the font, especially for the console and the elements panel. 

... I just checked out DevTools Author on Canary, with the built in dark theme + Author Monokai and that works quite well for me. Thanks!

**Comment by micjamking** [comment]
@okor awesome! Since you're using Canary, I'd definitely appreciate any feedback/bug reports you're noticing to help me get things cleaned up in preparation for default light/dark themes in Chrome stable. Feel free to leave an issue here and I'll try to follow up as soon as I can.

Mahalo!

**Comment by okor** [comment]
Yeah, it looks pretty rad overall. Only thing that stood out is the "sources" tab, where the sidebar ends up having a super low contrast - using the built in dark theme + Monokai from Author.

<img width="1144" alt="screen shot 2016-02-05 at 15 46 19" src="https://cloud.githubusercontent.com/assets/142910/12858690/a6ce5f3a-cc1f-11e5-8ca5-e2932e68414f.png">

And here it is flipped, using Monokai again (my go to for all editors) but with the Light built in theme.

<img width="1061" alt="screen shot 2016-02-05 at 15 50 20" src="https://cloud.githubusercontent.com/assets/142910/12858800/4323feb2-cc20-11e5-9ed6-12dc145a5b81.png">


For me personally, that is a non-issue as I actually don't normally write code in Chrome. The value for me is mostly having custom font's / themes in the markup (elements) and console. Which make the experience quite a bit nicer. Staring at bright screens all day is a pain, much prefer "dark themes" to most things.

But again, non issue for me. Sounds like it will be for you primary use case. Might have to add a couple of small overrides but largely should be about the same as you've been doing. Thanks again!

**Comment by micjamking** [comment]
@okor thanks for the feedback; I've opened a separate issue for the with the sources panel while using the dark theme in Canary. Feel free to add any other issues you find here: https://github.com/micjamking/devtools-author/issues/112
